core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -704,10 +707,22 @@ public void writeBatchToAllNonTxStores(Iterable<MarshallableEntry> entries,
       try {
          checkStoreAvailability();
          //noinspection unchecked
-         nonTxWriters.stream()
+         CompletableFuture batchFutures = CompletableFuture.allOf(
```

Comment:
This should be alright for now, I will change this if I can ever get to https://issues.jboss.org/browse/ISPN-9722